fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -502,6 +502,47 @@ private static SearchParameter getSearchParameterByUrlIfPresent(Map<String, Para
         return result;
     }
 
+    /**
+     * Perform wildcard processing for inclusion search parameters by getting all valid search parameters for the
+     * specified join resource type. Search parameters must have a type of 'reference'.
+     * <p>
+     * If inclusion keyword is ' _include' and a target resource type is specified, search parameter must contain
+     * a matching type.
+     * <p>
+     * If inclusion keyword is '_revinclude', search parameter must have a target resource type matching the resource
+     * type being searched.
+     * 
+     * @param resourceType
+     *            the resource type being searched for
+     * @param joinResourceType
+     *            the resource type for which inclusion search parameters will be returned
+     * @param searchParameterTargetType
+     *            the target resource type for included resources
+     * @param inclusionKeyword
+     *            the inclusion type, either _include or _revinclude
+     * @return
+     *         the inclusion SearchParameters for type {@code resourceType} or empty map if none exist
+     * @throws Exception
+     */
+    private static Map<String, SearchParameter> getInclusionWildcardSearchParameters(String resourceType, String joinResourceType,
+        String searchParameterTargetType, String inclusionKeyword) throws Exception {
+        Map<String, SearchParameter> inclusionSearchParameters = new HashMap<>();
+
+        for (SearchParameter searchParameter : getApplicableSearchParameters(joinResourceType)) {
+            if (searchParameter.getType().getValue().equals("reference") &&
+                    ((SearchConstants.INCLUDE.equals(inclusionKeyword)
+                            && (searchParameterTargetType == null || isValidTargetType(searchParameterTargetType, searchParameter))) ||
+                    (SearchConstants.REVINCLUDE.equals(inclusionKeyword) && isValidTargetType(resourceType, searchParameter)))) {
+                // Valid search parameter of type reference - add to map
+                inclusionSearchParameters.put(searchParameter.getCode().getValue(), searchParameter);
+            } else if (inclusionSearchParameters.containsKey(searchParameter.getCode().getValue())) {
+                // Search parameter is not valid - remove if search parameter by same name is alread in map
+                inclusionSearchParameters.remove(searchParameter.getCode().getValue());
```

Comment:
Why would there be two search parameters of the same code already in the map? 

The word already is misspelled

Reply:
This is covering a case where there's a built-in search parm of type 'reference', but there's a tenant-specific search parm of the same name and the type is not 'reference'. I thought the tenant-specific parm should win.

Reply:
Added log message and did not remove existing map entry.